internal/core/metadata/rest_deviceservice.go

```diff
@@ -639,39 +614,22 @@ func restUpdateServiceAdminStateByName(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Check if the device service exists
-	ds, err := dbClient.GetDeviceServiceByName(n)
-	if err != nil {
-		if err == db.ErrNotFound {
-			http.Error(w, "Device service not found", http.StatusNotFound)
-		} else {
+	op := device_service.NewUpdateAdminStateByNameExecutor(n, newAs, dbClient)
+	if err := op.Execute(); err != nil {
+		switch err.(type) {
+		case *types.ErrItemNotFound:
+			http.Error(w, err.Error(), http.StatusNotFound)
+		default:
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 		}
 		LoggingClient.Error(err.Error())
 		return
 	}
-
-	// Update the admins state
-	if err = updateServiceAdminState(ds, newAs, w); err != nil {
-		LoggingClient.Error(err.Error())
-		return
-	}
-	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
 	w.WriteHeader(http.StatusOK)
 	w.Write([]byte("true"))
 }
 
-// Update the admin state for the device service
-func updateServiceAdminState(ds models.DeviceService, as models.AdminState, w http.ResponseWriter) error {
-	ds.AdminState = as
-	if err := dbClient.UpdateDeviceService(ds); err != nil {
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
```

Comment:
I do not see this `503(Service Unavailable)` response code being used after these changes. Am I missing something or would these changes result in API changes?

For the record, I  think a `503` is not the correct response in this situation as it is meant to tell the client that the server is unavailable not that the entity does not exist.

Reply:
This changes the API, and I agree with your points.

The RAML doesn't describe a 503, so my best guess is that when we changed all the 503s to 500, we missed this one. We'll be rectifying that now and aligning with the RAML.